lib/launch-fastify.js

```diff
@@ -104,6 +104,14 @@ async function launchFastify(file, options, address) {
   const logger = customLogger(mergedOptions, options)
   const fastify = fastifyBuilder({ ...exportFastifyOptions(mergedOptions), logger })
 
+  const refResolver = mergedOptions.swaggerRefResolver || {
+    buildLocalReference(json, baseUri, fragment, i) {
+      if (process.env.npm_package_name && json.$id) {
+        return `${process.env.npm_package_name}-${json.$id}`
```

Comment:
why do we take the npm_package_name here?

Reply:
Yes, in this way if 2 microservices share a schema with the same id, they are not overridden

Reply:
I understand, however the fact that two microservices share the same id should not be handled here but in the component responsible for merging several OAS into a single one. If you have two services sharing the same package name this fix won't be enough.

Reply:
It's true, eventually the component which aggregates multiple swaggers should handle this, but at the moment we were told (by @ilteoood ) that it's better not to touch that component. 
Let's put it this way: this MR should lower the risk of having collision when multiple swaggers from multiple microservices have to be aggregated; in a second phase we should address the problem of handling collisions at the level of the swagger aggregator.

Reply:
The fact is that normally you use the same name in different definitions to refer to the same thing: using fastify-swagger this doesn't happen, because it assign random name, that will obviously collide.
Doing that on the merging component, will be too much slow.

Reply:
I understand that this solution mitigates the collition problem, however it does not properly address it. We could handle this in different ways, let's include @davidebianchi in the discussion to make sure we head in the right direction.

 * remove the default refResolver from the PR: still allow the definition for a refResolver (which surely is a nice feature), you'll have to define your own refResolver using the package name (or whatever suits you the best)
  * keep the mitigation here but also implement the fix in the aggregator component so that the problem is actually addressed

Reply:
I understand the problem, and I agree with the @fredmaggiowski point of view. 
The `fastify-swagger` default is `def-${i}`, which does not contain the `$id`. Why? Maybe could be some problems with same name at different levels?

For example, 
```js
function main(fastify) {
	instance.addSchema({ $id: 'Order', type: 'object', properties: { id: { type: 'integer' } } })
	instance.get('/foo', { schema: { body: { $ref: 'Order#' }, response: { 200: { $ref: 'Order#' } } } }, () => {})

  fastify.register(async (instance) => {
    instance.addSchema({ $id: 'Order', type: 'object', properties: { id: { type: 'integer' } } })
    instance.post('/', { schema: { body: { $ref: 'Order#' }, response: { 200: { $ref: 'Order#' } } } }, () => {})
  })
}
```
will work properly (I don't know if it is possible, maybe fastify will reject the duplication of id)?

If this works, we could set the default. Otherwise, I prefer to remove the default and overwrite it if needed (or change the swagger-aggregator)

Reply:
Yes, in the example made be @davidebianchi there would be collisions.

For this reason I will remove the default refResolver and leave the feature of defining a custom refResolver.